python/tvm/relay/ir_pass.py

```diff
@@ -1,16 +1,292 @@
 # pylint: disable=no-else-return
 # pylint: disable=unidiomatic-typecheck
-"""The set of passes for Relay.
+"""
+This file contains:
+1. The set of passes for Relay, which exposes an interface for configuring the
+   passes and scripting them in Python.
 
-Exposes an interface for configuring the passes and
-scripting them in Python.
+2. The pass manager for Relay which exposes different granularity of interfaces
+   for users to implement and use passes more conveniently.
 """
+import types
+
 from . import _ir_pass
 from . import _make
 from .expr import Expr
 from .ty import Type
+from .base import RelayNode, register_relay_node
 from .module import Module
 
+
+@register_relay_node
+class PassInfo(RelayNode):
+    """The class that contains the meta data required by a pass. It is the
+    container of information needed by running an optimization or analysis.
+    This class can be extended by adding new members when more meta data is
+    needed.
+
+    Parameters
+    ----------
+    name : str
+        The pass name.
+
+    opt_level : int
+        The optimization level of this pass.
+
+    required : List[str]
+        The list of passes that are required by a certain pass.
+    """
+
+    def __init__(self, name, opt_level, required=None):
+        self.__init_handle_by_constructor__(_ir_pass.PassInfo, name, opt_level,
+                                            required)
+
+
+@register_relay_node
+class PassContext(RelayNode):
+    """The basis where a Relay optimization/analysis runs on.
+    Each pass context contains a number of auxiliary information that is used
+    to help an optimization pass. Such information includes the error reporter
+    to record the errors of during the optimization, etc.
+    """
+
+    def __init__(self):
+        self.__init_handle_by_constructor__(_ir_pass.PassContext)
+
+
+@register_relay_node
+class Pass(RelayNode):
+    """The base class of all passes. All methods here are just simple wrappers
+    that are implemented in the backend. They are defined for users to
+    conveniently interact with the base class.
+    """
+
+    def set_pass_context(self, pass_ctx):
+        """Setup the pass context for analysis and optimizations. This context
+        could be shared by different passes for sequential passes.
+
+        Parameters
+        ----------
+        pass_ctx : PassContext
+            The context that is used to help perform a certain pass or a series
+            of passes.
+        """
+        if not isinstance(pass_ctx, PassContext):
+            raise TypeError("pass_ctx is expected to be the PassContext type")
+        _ir_pass.SetContext(self, pass_ctx)
+
+    def get_pass_info(self):
```

Comment:
```
@property
def info(self):
   ...
```

Reply:
I tried to use property here. But it looks that I cannot access it through the derived classes because it says property returns an object.